H2TauTau/python/heppy/analyzers/FakeFactorAnalyzer.py

```diff
@@ -0,0 +1,152 @@
+        return ff.value(len(inputs), array('d',inputs))
+
+    def set_ff_fullyhadronic(self, tau, tau2, njets, mvis):
+        for cat in ["btag","nobtag","inclusive"]:
+            setattr(tau, 
+                    'weight_fakefactor_{}',
+                    self.fake_factor_fullyhadronic(tau,
+                                                   tau2,
+                                                   njets,
+                                                   mvis))
+            setattr(tau, 
+                    'weight_fakefactor_{}_up',
+                    self.fake_factor_fullyhadronic(tau,
+                                                   tau2,
+                                                   njets,
+                                                   mvis,
+                                                   'up'))
+            setattr(tau, 
+                    'weight_fakefactor_{}_down',
+                    self.fake_factor_fullyhadronic(tau,
+                                                   tau2,
+                                                   njets,
+                                                   mvis,
+                                                   'down'))
+
+    def set_ff_semileptonic(self, tau,  njets, mvis, mt, iso):
+        for cat in ["btag","nobtag","inclusive"]:
+            setattr(tau, 
+                    'weight_fakefactor_{}',
+                    self.fake_factor_semileptonic(tau,
+                                                  njets,
+                                                  mvis,
+                                                  mt,
+                                                  iso))
+            setattr(tau, 
+                    'weight_fakefactor_{}_up',
+                    self.fake_factor_semileptonic(tau,
+                                                  njets,
+                                                  mvis,
+                                                  mt,
+                                                  iso,
+                                                  'up'))
+            setattr(tau, 
+                    'weight_fakefactor_{}_down',
+                    self.fake_factor_semileptonic(tau,
+                                                  njets,
+                                                  mvis,
+                                                  mt,
+                                                  iso,
+                                                  'down'))
+
+    def process(self, event):
+        if not self.cfg_comp.isData:
+            return True
+
+        njets = len(event.jets30)
+        mvis = event.dileptons_sorted[0].mass()
+
+        if self.cfg_ana.channel == 'tt':
+            tau1 = event.dileptons_sorted[0].leg1()
+            tau2 = event.dileptons_sorted[0].leg2()
+
+            self.set_ff_fullyhadronic(tau1, tau2, njets, mvis)
+            self.set_ff_fullyhadronic(tau2, tau1, njets, mvis)
+
```

Comment:
remove linebreak (avoid them when possible and especially in an if else structure)